src/osp/Active/ActiveScene.cpp

```diff
@@ -39,196 +40,41 @@ void ACompCamera::calculate_projection()
 }
 
 
-ActiveScene::ActiveScene(UserInputHandler &userInput, OSPApplication &app, Package& context) :
+ActiveScene::ActiveScene(OSPApplication &app, Package& context) :
         m_app(app),
-        m_context(context),
-        m_hierarchyDirty(false),
-        m_userInput(userInput)
+        m_context(context)
 {
-    m_registry.on_construct<ACompHierarchy>()
-                    .connect<&ActiveScene::on_hierarchy_construct>(*this);
-
-    m_registry.on_destroy<ACompHierarchy>()
-                    .connect<&ActiveScene::on_hierarchy_destruct>(*this);
-
     // Create the root entity
-
     m_root = m_registry.create();
-    ACompHierarchy& hierarchy = m_registry.emplace<ACompHierarchy>(m_root);
-    hierarchy.m_name = "Root Entity";
-
 }
 
 ActiveScene::~ActiveScene()
 {
     m_registry.clear();

```

Comment:
Remind me why we need to call .clear()? the ActiveReg_t should be destructed here. Isn't that enough?

Reply:
Unfortunately, I can't really pin down why, but ACompDebugObject and DebugCameraController causes a crash on exit when this isn't here. The stack trace shows the error happening in a seemingly unrelated spot that seems to change when small parts of the code are changed. Clearly a sign of undefined behaviour going on.

It probably has something to do with the order at which ActiveScene's members are destructed. Something that DebugCameraController needs.

Reply:
Ok, figured it out. It's the unreliable FunctionOrderHandle causing the issues. All of the UpdateOrderHandle_t and RenderOrderHandle_t have to be destructed first, or else they write into an already destructed FunctionOrder messing up something else.

Reply:
I mentioned that FunctionOrder is so incredibly unreliably, that one should "hold a gun up to it at all times." Maybe I should actually shoot it next PR. Calling all the system update functions explicitly wouldn't really be a bad idea for the time being.